packages/interactivity/src/init.js

```diff
@@ -21,15 +21,26 @@ export const getRegionRootFragment = ( region ) => {
 	return regionRootFragments.get( region );
 };
 
+function yieldToMain() {
+	return new Promise( ( resolve ) => {
+		// TODO: Use scheduler.yield() when available.
+		setTimeout( resolve, 0 );
+	} );
+}
+
 // Initialize the router with the initial DOM.
 export const init = async () => {
-	document
-		.querySelectorAll( `[data-${ directivePrefix }-interactive]` )
-		.forEach( ( node ) => {
-			if ( ! hydratedIslands.has( node ) ) {
-				const fragment = getRegionRootFragment( node );
-				const vdom = toVdom( node );
-				hydrate( vdom, fragment );
-			}
-		} );
+	const nodes = document.querySelectorAll(
+		`[data-${ directivePrefix }-interactive]`
+	);
+
+	for ( const node of nodes ) {
+		if ( ! hydratedIslands.has( node ) ) {
+			await yieldToMain();
+			const fragment = getRegionRootFragment( node );
+			const vdom = toVdom( node );
+			await yieldToMain();
```

Comment:
@luisherranz Actually, what is the purpose of `hydratedIslands` here? Could it be that the check for `! hydratedIslands.has( node )` could be done so that the condition is entered, but then yielding occurs so that it's possible that the island could be hydrated by another task? Could it be that `vDom()` accidentally gets called on a hydrated island?

Reply:
> what is the purpose of hydratedIslands here?

Islands that are inside other islands belong to the same _Preact application_ so they can share context and such. When an island is already hydrated as part of a parent, we skip its own hydration.

> that it's possible that the island could be hydrated by another task?

I actually [spotted a bug yesterday](https://github.com/WordPress/wordpress-develop/pull/5922#issuecomment-1910673494) due to the fact that the router store was calling `toVdom` before the `init`. It's a rare edge case, but I'll fix it.

As `toVdom` is going to be a private, core-only API in WP 6.5, there should no be more problems. But in the future, if `toVdom` is exposed publicly, we can add a second argument to `toVdom( node, isHydrating = false )` to know if it should add the nodes to the `hydratedIslands` map or not.